internal/config/datasources.go

```diff
@@ -0,0 +1,39 @@
+package config
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/anchore/grype/grype/matcher/java"
+)
+
+const (
+	defaultMavenBaseURL = "https://search.maven.org/solrsearch/select"
+)
+
+type externalSources struct {
```

Comment:
"It seems if there are additional external sources we add, having them all mixed together here might not be ideal."

I think that's when we do the refactor. I think since this is the first PR with anything akin to external sources it's ok to have it under one file and then reorganize it as we learn more about the shape that this feature will take. I don't want to organize it a certain way and make choices for future authors that could presume a more advanced config design. 

"I think having external sources be an all-or-nothing thing (e.g. external-sources.enable"
This was specifically asked for to have an on off master switch.

If that's enabled THEN we check the individual configs.

Reply:
@wagoodman @luhring @jonasagx thoughts here?

Reply:
> This was specifically asked for to have an on off master switch.
👍 

Agree solving for all the cases is unnecessary until there are more than one.